core/src/main/java/com/linecorp/armeria/client/http/DefaultHttpClient.java

```diff
@@ -78,4 +85,27 @@ HttpResponse execute(EventLoop eventLoop, AggregatedHttpMessage aggregatedReq) {
         req.close();
         return execute(eventLoop, req);
     }
+
+    private static String concatPath(String path1, String path2) {
```

Comment:
This method is too heavyweight. What I would expect is just a simple stuff like this:

```java
private static String concatPath(String path1, String path2) {
  path1 = path1 == null ? "" : path1;
  path2 = path2 == null ? "" : path2;

  final String concatPath;
  if (path1.isEmpty()) {
    if (path2.isEmpty()) {
        concatPath = "/";
    } else if (path2.charAt(0) == '/') {
        concatPath = path2;
    } else {
        concatPath = '/' + path2;
    }
  } else if (path2.isEmpty()) {
    concatPath = path1;
  } else if (path1.charAt(path1.length() - 1) == '/') {
    if (path2.charAt(0) == '/') {
      concatPath = new StringBuilder(path1.length() + path2.length() - 1).append(path1).append(path2, 1, path2.length()).toString();
    } else {
      concatPath = path1 + path2;
    }
  } else {
    if (path2.charAt(0) == '/') {
      concatPath = path1 + path2;
    } else {
      concatPath = new StringBuilder(path1.length() + path2.length() +1).append(path1).append('/').append(path2).toString();
    }
  }
}
```